src/app/modules/react-migration/react-migration.controller.ts

```diff
@@ -38,8 +43,86 @@ export const ADMIN_COOKIE_OPTIONS_WITH_EXPIRY = {
 
 const logger = createLoggerWithLabel(module)
 
-const serveFormReact: ControllerHandler = (_req, res) => {
-  const reactFrontendPath = path.resolve('dist/frontend')
+const reactFrontendPath = path.resolve('dist/frontend')
+const reactHtml = readFileSync(path.join(reactFrontendPath, 'index.html'), {
+  encoding: 'utf8',
+})
+
+type MetaTags = {
+  title: string
+  description: string
+  image: string
+}
+const replaceWithMetaTags = ({
+  title,
+  description,
+  image,
+}: MetaTags): string => {
+  return (
+    reactHtml
+      // there are multiple __OG_TITLE__ and we are unable to use string.replaceAll so we use regexp
+      .replace(new RegExp('__OG_TITLE__', 'g'), title)
+      .replace('<title>FormSG</title>', `<title>${title}</title>`)
+      .replace('__OG_DESCRIPTION__', description)
+      .replace('__OG_IMAGE__', image)
+  )
+}
+
+const serveFormReact =
+  (isPublicForm: boolean): ControllerHandler =>
+  (req, res, next) => {
+    const reactFrontendPath = path.resolve('dist/frontend')
+    logger.info({
+      message: 'serveFormReact',
+      meta: {
+        action: 'routeReact.serveFormReact',
+        __dirname,
+        cwd: process.cwd(),
+        reactFrontendPath,
+      },
+    })
+
+    if (isPublicForm && !!get(req.params, 'formId')) {
```

Comment:
/nit Unless the linter is complaining, the `!!` is not necessary in a pure evaluation (i.e. the evaluation is not being assigned to a variable of type boolean)